apps/client/src/app/components/user-account-settings/user-account-settings.component.ts

```diff
@@ -99,6 +113,31 @@ export class UserAccountSettingsComponent implements OnDestroy, OnInit {
     return !(this.language === 'de' || this.language === 'en');
   }
 
+  public onCloseAccount() {
+    const confirmation = confirm(
+      $localize`Do you really want to close your account?`
+    );
+
+    const accessToken = this.deleteOwnUserForm.get('accessToken').value;
+
+    if (confirmation) {
+      this.dataService
+        .deleteOwnUser(accessToken)
+        .pipe(takeUntil(this.unsubscribeSubject))
+        .subscribe({
+          next: () => {
+            this.tokenStorageService.signOut();
```

Comment:
Can you extract

```
this.tokenStorageService.signOut();
this.userService.remove();

document.location.href = `/${document.documentElement.lang}`;
```

to a new function `this.userService.signOut()` and also use it from `AppComponent`?

Reply:
Create userService.signOut raises issue: "Circular dependency in DI detected". 
TokenStorageService already depends on UserService to set the userState to null when user sign out. 
Should leave it as it is and create issue  for  this?

Reply:
Ah okay, then leave it as is for now.